sycl/source/detail/queue_impl.hpp

```diff
@@ -447,6 +482,9 @@ class queue_impl {
   // Thread pool for host task and event callbacks execution.
   // The thread pool is instantiated upon the very first call to getThreadPool()
   std::unique_ptr<ThreadPool> MHostTaskThreadPool;
+
+  // Buffer to store assert failure descriptor
+  buffer<AssertHappened, 1> MAssertHappenedBuffer;
```

Comment:
Minor. Can we have one buffer per context?

Reply:
A single context can be for multiple devices and I'd like to avoid cross-device copies here.

Reply:
Why? I'm not sure having multiple devices in the same context means need to do cross-device copies.
Anyway, we could you have a buffer per each device in the context.
PS. I'm not asking to make any changes around this area in this PR.